lib/elixir/lib/enum.ex

```diff
@@ -1115,7 +1115,10 @@ defmodule Enum do
   @spec frequencies(t) :: map
   def frequencies(enumerable) do
     reduce(enumerable, %{}, fn key, acc ->
-      Map.update(acc, key, 1, &(&1 + 1))
+      case acc do
+        %{^key => value} -> %{acc | key => value + 1}
+        %{} -> Map.put(acc, key, 1)
```

Comment:
I think we can avoid pattern matching on `${}` since `acc` will always be a map., we can replace it with `_`

Reply:
OK I see

Reply:
It is fine as is. I prefer to write code in the style above, as it is clearer in intent.